pkg/integrations/v4/dm/id_provider.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dm
+
+import (
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/pkg/backend/identityapi"
+	"github.com/newrelic/infrastructure-agent/pkg/entity"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
+)
+
+type RegisteredEntitiesNameToID map[string]entity.ID
+type UnregisteredEntitiesNamed map[string]UnregisteredEntity
+type reason string
+
+type UnregisteredEntity struct {
+	Reason reason
+	Err    error
+	Entity protocol.Entity
+}
+
+type UnregisteredEntities []UnregisteredEntity
+
+const reasonClientError = "Identity client error"
+const reasonEntityError = "Entity error"
+
+func newUnregisteredEntity(entity protocol.Entity, reason reason, err error) UnregisteredEntity {
+	return UnregisteredEntity{
+		Entity: entity,
+		Reason: reason,
+		Err:    err,
+	}
+}
+
+type idProviderInterface interface {
```

Comment:
id provider is a cross cutting concern, we might want to extract it from `dm` as it does not only apply to dimensional metrics

Reply:
I agree but since is being used by dm_emitter to avoid circular references we move it to this package.

And at the end dm it will be the only one that use it.